internal/common/tls.go

```diff
@@ -0,0 +1,45 @@
+package common
+
+var defaultTLSCipherSuites = []string{
+	"TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
+	"TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256",
+	"TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
+	"TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256",
+	"TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
+	"TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
+}
+
+func DefaultTLSCipherSuites() []string {
+	return defaultTLSCipherSuites
+}
+
+// NOTE: the following cipher suites are currently supported by Envoy but insecure and
```

Comment:
Note that we might have some issues here since we allow specifying which version of envoy you want to use for a gateway as part of the `GatewayClassConfig` CRD. Not that we can exactly handle version checking at the controller level since you could even specify a custom envoy container/build, but we may want to document in the actual docs that this option is potentially particularly sensitive to the envoy version that is being deployed for the gateway.